src/public/js/rs.js

```diff
@@ -45,13 +45,214 @@ var RS = (function() {
       }
     }
     if (errors) {
-      // TODO ERROR CORRECTION
-      throw new Error('Error correction not implemented yet');
+      // Perform B-M decoding
+      var syndromes = computeSyndromePoly(receivedMsg, n, k);
+      var bm = berlekampMassey(syndromes, n, k);
+      var errorLocations = chienSearch(bm.errorLocationPoly);
+      var errorMagnitudes = forneysFormula(errorLocations,
+                                           bm.errorMagnitudePoly);
+      var repairedMsg = repair(receivedMsg, errorLocations, errorMagnitudes);
+
+      // Check that the repaired message is in fact divisible by genPoly
+      var repairedRemainder = UTIL.polynomialDiv(repairedMsg, genPoly, n, k);
+      var repairedErrors = false;
+      for (var i = 0; i < n - k; i++) {
+        if (repairedRemainder[i] != 0) {
+          throw new Error('Too many errors to decode.');
+        }
+      }
+
+      return repairedMsg.slice(0, k);
     } else {
-      return codedMsg.slice(0, k);
+      return receivedMsg.slice(0, k);
+    }
+  }
+
+  // Computes a syndrome polynomial for the specified received message
```

Comment:
Possibly change comment to something like "computes syndromes and represents them as a polynomial Si*X^i or something since the syndromes are actually a set of points and not a polynomial by nature.